dashboard/app/controllers/api_controller.rb

```diff
@@ -153,6 +153,105 @@ def section_text_responses
     render json: data
   end
 
+  def section_assessments
+    load_section
+    load_script
+
+    level_group_script_levels = @script.script_levels.includes(:level).where('levels.type' => LevelGroup)
+
+    data = @section.students.map do |student|
+      student_hash = {id: student.id, name: student.name}
+
+      level_group_script_levels.map do |script_level|
+        next unless script_level.long_assessment?
+
+        last_attempt = student.last_attempt(script_level.level)
+        response = last_attempt.try(:level_source).try(:data)
+
+        next unless response
+
+        user_level = UserLevel.where(user_id: student.id, level: script_level.level, script: @script).first
+        puts user_level.to_json
+
+        # Summarize some key data.
+        multi_count = 0
+        multi_count_correct = 0
+
+        # And construct a listing of all the individual levels and their results.
+        level_results = []
+
+        next unless properties = script_level.level[:properties]
+
+        # Go through each page of the long assessment's LevelGroup.
+        properties["pages"].each do |page|
+          # Go through each level of the page.
+          page["levels"].each do |page_level_name|
+            level = Level.find_by_name(page_level_name)
+
+            # And work with the response to the individual level inside the LevelGroup.
+            response_parsed = JSON.parse(response)
+            level_response = response_parsed.select{|r| r["level_id"] == level.id}
+
+            if level_response
+              # For each level inside the LevelGroup, we determine the resut,
+              # both content in :student_result and result value in :correct.
+              # This array is returned for each LevelGroup, and its order matches
+              # that of the LevelGroup as it's declared.
+              level_result = {}
+
+              level_class = level.class.to_s.underscore
```

Comment:
since you only use this string for comparisons, you might as well compare the class directly, a couple ways to do this: 

* `if level.is_a? TextMatch`
* `if level.instance_of? TextMatch` http://stackoverflow.com/questions/3893278/ruby-kind-of-vs-instance-of-vs-is-a instance_of returns true for the exact class while is_of allows subclasses. May want to use instance_of here because the inheritance is kind of weird (like we have `Multi < Match`)
* `if TextMatch === level` The `===` might read weird to people especially if they are used to using `===` in JS and because it's 'backwards'.
* 
````
case level
when TextMatch
when Multi
````
actually uses === in the same way as above; I think looks better especially if there may be more than 2 conditions in the future